sphinx/writers/latex.py

```diff
@@ -424,21 +424,32 @@ def escape_abbr(text):
     return re.sub(r'\.(?=\s|$)', r'.\@', text)
 
 
-def rstdim_to_latexdim(width_str):
-    # type: (unicode) -> unicode
+def rstdim_to_latexdim(width_str, scale = 100):
+    # type: (unicode, int) -> unicode
     """Convert `width_str` with rst length to LaTeX length."""
     match = re.match(r'^(\d*\.?\d*)\s*(\S*)$', width_str)
     if not match:
         raise ValueError
     res = width_str
     amount, unit = match.groups()[:2]
-    float(amount)  # validate amount is float
-    if unit in ('', "px"):
-        res = "%s\\sphinxpxdimen" % amount
-    elif unit == 'pt':
-        res = '%sbp' % amount  # convert to 'bp'
-    elif unit == "%":
-        res = "%.3f\\linewidth" % (float(amount) / 100.0)
+    if scale == 100:
+        float(amount)  # validate amount is float
+        if unit in ('', "px"):
+            res = "%s\\sphinxpxdimen" % amount
+        elif unit == 'pt':
+            res = '%sbp' % amount  # convert to 'bp'
+        elif unit == "%":
+            res = "%.3f\\linewidth" % (float(amount) / 100.0)
```

Comment:
In future `%.3f` when merging to 2.0 should be replaced by `%.5f` because TeX has about 5 decimal digits precision in such computations.

For time being I am keeping exact same code as we are on stable branch.